Automattic/wp-calypso
# composite-checkout: Add thank-you page decision making

#### Changes proposed in this Pull Request

When a payment is complete, there are a myriad of thank-you pages available to be redirected to. Here are the options that I'm aware of, as they are defined in the page controller; there may be more.

- /checkout/thank-you/no-site/pending/:orderId
- /checkout/thank-you/no-site/:receiptId?
- /checkout/thank-you/:site/pending/:orderId
- /checkout/thank-you/:site/:receiptId?
- /checkout/thank-you/:site/:receiptId/with-gsuite/:gsuiteReceiptId
- /checkout/thank-you/features/:feature/:site/:receiptId?

This PR adds a new function, `getThankYouPageUrl`, which will be the canonical place to determine which url to use.

#### Testing instructions

- You can run the unit tests with `npm run test-client client/my-sites/checkout/checkout/test/composite-checkout-thank-you`
- Run calypso with `ENABLE_FEATURES=composite-checkout-wpcom npm start`.
- Using an account with only a single site, add a plan to your cart and complete checkout. Verify that you are redirected to a thank-you page that includes that site slug or id in the url.
- Repeat the above step with an account that has multiple sites.
- Repeat the above step except purchase "just a domain" by starting at the `/domains` url. Verify that the thank-you page includes `no-site` in the url.
- If you have time, try a different payment method to make sure different ones work. PayPal, notably, works differently.
